umap/settings/docker.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding:utf-8 -*-
+"""
+Settings for Docker development
+
+Use this file as a base for your local development settings and copy
+it to umap/settings/local.py. It should not be checked into
+your code repository.
+"""
+import environ
+from umap.settings.base import *   # pylint: disable=W0614,W0401
+
+env = environ.Env()
+
+SECRET_KEY = env('SECRET_KEY')
+INTERNAL_IPS = env.list('INTERNAL_IPS', default='127.0.0.1')
+ALLOWED_HOSTS = env.list('ALLOWED_HOSTS', default='*')
+
+DEBUG = env.bool('DEBUG', default=False)
+
+ADMIN_EMAILS = env.list('ADMIN_EMAIL', default='')
+ADMINS = [(email, email) for email in ADMIN_EMAILS]
+MANAGERS = ADMINS
+
+DATABASES = {
+    'default': env.db(default='postgis://localhost:5432/umap')
+}
+
+COMPRESS_ENABLED = True
+COMPRESS_OFFLINE = True
+
+LANGUAGE_CODE = 'en'
+
+# Set to False if login into django account should not be possible. You can
+# administer accounts in the admin interface.
+ENABLE_ACCOUNT_LOGIN = env.bool('ENABLE_ACCOUNT_LOGIN', default=True)
+
+AUTHENTICATION_BACKENDS = ()
+
+# We need email to associate with other Oauth providers
+SOCIAL_AUTH_GITHUB_SCOPE = ['user:email']
+SOCIAL_AUTH_GITHUB_KEY = env('GITHUB_KEY', default='')
+SOCIAL_AUTH_GITHUB_SECRET = env('GITHUB_SECRET', default='')
+if SOCIAL_AUTH_GITHUB_KEY and SOCIAL_AUTH_GITHUB_SECRET:
+    AUTHENTICATION_BACKENDS += (
+        'social_core.backends.github.GithubOAuth2',
+    )
+SOCIAL_AUTH_BITBUCKET_KEY = env('BITBUCKET_KEY', default='')
+SOCIAL_AUTH_BITBUCKET_SECRET = env('BITBUCKET_SECRET', default='')
+if SOCIAL_AUTH_BITBUCKET_KEY and SOCIAL_AUTH_BITBUCKET_SECRET:
+    AUTHENTICATION_BACKENDS += (
+        'social_core.backends.bitbucket.BitbucketOAuth',
+    )
+
+SOCIAL_AUTH_TWITTER_KEY = env('TWITTER_KEY', default='')
+SOCIAL_AUTH_TWITTER_SECRET = env('TWITTER_SECRET', default='')
+if SOCIAL_AUTH_TWITTER_KEY and SOCIAL_AUTH_TWITTER_SECRET:
+    AUTHENTICATION_BACKENDS += (
+        'social_core.backends.twitter.TwitterOAuth',
+    )
+SOCIAL_AUTH_OPENSTREETMAP_KEY = env('OPENSTREETMAP_KEY', default='')
+SOCIAL_AUTH_OPENSTREETMAP_SECRET = env('OPENSTREETMAP_SECRET', default='')
+if SOCIAL_AUTH_OPENSTREETMAP_KEY and SOCIAL_AUTH_OPENSTREETMAP_SECRET:
```

Comment:
Out of scope for this PR, but someone may want to run uMap with another OAuth provider. Not sure how to make this easy/possible with Docker.